src/linear_algebra/bilinear_form.lean

```diff
@@ -710,6 +732,73 @@ end to_matrix
 
 end matrix
 
+namespace finsupp
+
+variables {α β : Type*}
+variables {M' : Type*} [add_comm_monoid M']
+variables {N : Type*} [has_zero N]
+
+lemma bilin_add_left' (A : α → β → N → (M' →+ R)) (p q : α →₀ M') (r : β →₀ N) :
+  (p + q).sum (λ i x, r.sum (λ j y, A i j y x))
+  = p.sum (λ i x, r.sum (λ j y, A i j y x)) + q.sum (λ i x, r.sum (λ j y, A i j y x)) :=
```

Comment:
I wonder if a `finsupp.sum₂` definition would be useful here - It could be used here too:

https://github.com/leanprover-community/mathlib/blob/07fc9821f41bd369d930fa9ff8e83e56935e7594/src/algebra/monoid_algebra.lean#L72-L73

and maybe would share some lemmas

Reply:
Now that we have a better big operator library, I would rather get rid of `sum` entirely:
```lean
instance : has_mul (monoid_algebra k G) :=
⟨λ f g, ∑ i in f.support, ∑ j in g.support, single (i * j) (f i * g j)⟩
```

> maybe would share some lemmas

This is the reason why I'd rather have one definition, and a third of the lemmas.

Reply:
I agree that `finsupp.sum` could be profitably replaced with `finset.sum`, or even `finsum`. Let's stick to `finsupp.sum` in this PR, since the linear algebra library uses it consistently throughout (for now).